app/analyzers/metrics.py

```diff
@@ -223,6 +236,46 @@ def calculate_metric(metric, value):
 
             return extracted_urls_length, observations
 
+        # ---------------------------------------------------------
+        # METRIC: Compare entropy of observed URL's with entropy of the majestic million domains
+        # The more a URL diverges from those seen in the majestic million, the higher this entropy value will be
+        # ---------------------------------------------------------
+        elif metric == "relative_url_entropy":
+            majestic_million_frequencies = \
+                {'g': 0.02706810814315049, 'o': 0.07421531631063037, 'l': 0.04660619075683699, 'e': 0.0938650686651803,
+                 'f': 0.016087468884472687, 'a': 0.08965206537963542,'c': 0.046178435692422186, 'b': 0.021492396761465096,
+                 'k': 0.017011742091988323,'y': 0.017683540507870608, 'u': 0.03113815167654972, 't': 0.05877603780957555,
+                 'w': 0.012812697524051385, 'i': 0.07074249978897978, 'r': 0.06343497059722608,'m': 0.033597415407595026,
+                 's': 0.06260194430883878, 'n': 0.06262892491736954, 'd': 0.031030885021106236,'p': 0.026214752715696614,
+                 'v': 0.013545577039801925, 'h': 0.027979827873085842, 'z': 0.007096836870275642,'-': 0.010803953745868712,
+                 '3': 0.0020435937308682425, 'q': 0.002948193577996864, 'x': 0.006551510056881306,'j': 0.006711051641353142,
+                 '0': 0.0027525841661488358, '1': 0.0029865097894172872,'2': 0.0027598914142925837, '6': 0.0017662930320798498,
+                 '4': 0.001964712923983166,'5': 0.0017608594373062934, '8': 0.0021597602398201366, '9': 0.0017470880850353834,
+                 '7': 0.0015831434151435972}
+
+            extracted_urls = []
+
+            # if the target value is a list of strings, convert it into a single list of strings
+            target_value_words = value.replace('"', ' ').split()  # splits on whitespace by default, and on quotes, since we most likely will apply this to parameter arguments
+
+            for word in target_value_words:
+                is_url = helpers.utils.is_url(word)
+                if is_url:
+                    extracted_urls.append(word)
+
+                observations["extracted_urls"] = ','.join(extracted_urls)
```

Comment:
I don't think you want ta add extracted_urls if none were observed, so I guess you need to put this in a conditional clause which checks actual URL's were extracted